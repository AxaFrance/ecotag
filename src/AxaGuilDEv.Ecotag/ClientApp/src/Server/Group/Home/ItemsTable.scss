.af-table {
  &__small {
    width: 25px;
  }
}