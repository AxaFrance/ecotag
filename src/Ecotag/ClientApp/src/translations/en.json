{
  "header": {
    "title": "An eco-system that aims to be eco-friendly",
    "errors": {
      "not_connected": "Not connected"
    }
  },
  "paging": {
    "display": "Display",
    "elements": "elements",
    "previous": "« Previous",
    "next": "Next »"
  },
  "stepper": {
    "confirm": "Confirm"
  },
  "resilience": {
    "error": {
      "first_part": "An error occurred.",
      "second_part": "We are aware of the problem. An investigation is ongoing..."
    },
    "forbidden": "You are not allowed to access this resource."
  },
  "not_found": {
    "title": "404 Page Not Found",
    "not": "not",
    "found": "found",
    "message": "The page you are looking for is not here !"
  },
  "home": {
    "title": "Home",
    "projects": "Projects",
    "datasets": "Datasets",
    "groups": "Teams",
    "errors": {
      "no_roles": "No role is associated to your profile."
    }
  },
  "group": {
    "title": "Teams",
    "subtitle": "Annotation teams governance",
    "creation_title": "Creation",
    "new": {
      "name_input": {
        "label": "Name : ",
        "help_message": "Ex : bank account details annotators"
      },
      "validation": "Validate"
    },
    "count_teams": {
      "first_part": "There is/are (",
      "second_part": ") team(s)"
    },
    "empty_list": "No element",
    "list": {
      "header": {
        "name": "Name",
        "users": "Users",
        "actions": "Actions"
      },
      "edit": "Edit",
      "edit_popup": {
        "header": "Add/Remove users from this team",
        "add_user_placeholder": "Add user",
        "footer": {
          "cancel": "Cancel",
          "validation": "Validate"
        }
      }
    }
  },
  "dataset": {
    "title": "Datasets",
    "subtitle": "A dataset is the representation of a collection of data",
    "new_dataset": "New dataset",
    "access_token": "Access token",
    "active_datasets_counter": "Your active datasets : ",
    "filter_by": "Filter by",
    "name": "Name",
    "empty_list": "No elements",
    "list": {
      "actions_titles": {
        "locked": "locked",
        "locked_and_work_in_progress": "locked and work in progress",
        "pending": "pending",
        "none": "unlocked"
      },
      "header": {
        "lock": "Lock",
        "name": "Name",
        "team": "Team",
        "classification": "Classification",
        "nb_files": "Number of files",
        "type": "Type",
        "creation_date": "Creation date",
        "action": "Action"
      },
      "edit": "Edit"
    },
    "new": {
      "title": "New dataset",
      "stepper": {
        "title": "New dataset"
      },
      "form_title": "New dataset",
      "name_input": {
        "label": "Name",
        "help_message": "Ex : bank-account-details-dataset"
      },
      "team_input": {
        "label": "Team"
      },
      "type_input": {
        "label": "Type",
        "help_message": "Ex : Image, Text",
        "options_labels": {
          "image": "Images (.jpg, .png, jpeg)",
          "document": "Documents (.txt, .eml, .tiff, .pdf, .doc, etc.)",
          "text": "Text (.txt)",
          "eml": "Emails (.eml)"
        },
        "help": {
          "title": "Information on the accepted documents types",
          "images": {
            "name": "Images",
            "types": " : .jpg, .png, jpeg"
          },
          "documents": {
            "name": "Documents",
            "types": " : .txt, .eml, .msg, .jpg, .png, jpeg, .tiff, .tif, .pdf, .doc, .docx, .xls, .xlsx, .ppt, .pptx, .rtf, .odt, .ods, .odp, .zip"
          },
          "text": {
            "name": "Text",
            "types": " : .txt"
          },
          "eml": {
            "name": "Eml",
            "types": " : .eml"
          }
        }
      },
      "classification_input": {
        "label": "Classification",
        "help_message": "Ex : Critical",
        "options": {
          "public": {
            "label": "Public"
          },
          "internal": {
            "label": "Internal"
          },
          "confidential": {
            "label": "Confidential"
          },
          "critical": {
            "label": "Critical"
          }
        },
        "help": {
          "title": "Information on criticality",
          "public": {
            "name": "Public",
            "detail": " : a piece of information made deliberately for the public, or specifically made for public diffusion"
          },
          "internal": {
            "name": "Internal",
            "detail": " : a piece of information that is accessible to all or a large portion of employees and is not intended for external parties."
          },
          "confidential": {
            "name": "Confidential",
            "detail": " : a piece of information restricted to a specific audience"
          },
          "critical": {
            "name": "Critical",
            "detail": " : Please refer to your manager."
          }
        }
      },
      "import_input": {
        "label": "Import from a blob : "
      },
      "dataset_to_import": {
        "label": "Dataset to import : "
      },
      "validate": "Validation"
    },
    "confirm": {
      "title": "Confirmation",
      "stepper": "New dataset",
      "confirmation_title": "New dataset added !",
      "confirmation_description": "If you have imported a dataset, the process of adding files can take a while.",
      "return_button_label": "Go back to datasets list"
    },
    "edit": {
      "confirm_modal": {
        "title": "Do you want to lock the dataset permanently ?",
        "warning_first_part": "This action is permanent.",
        "warning_second_part": "Every modification (addition or deletion of files) will be impossible then."
      },
      "subtitle": "Dataset edition",
      "lock": "Lock",
      "locked_state": {
        "locked": "Dataset locked",
        "locked_and_work_in_progress": "Dataset is being worked on",
        "default": "Dataset is being imported"
      },
      "file_upload": {
        "title": "Upload files ",
        "file_input": {
          "label": "Browse",
          "placeholder": "Drag/drop your files here"
        },
        "send_button": {
          "label": "Send"
        },
        "errors": {
          "already_existing": {
            "first_part": "The file ",
            "second_part": " already exists"
          },
          "could_not_upload": {
            "first_part": "The file ",
            "second_part": " could not be uploaded"
          }
        }
      },
      "files_list": {
        "global_info_tab": {
          "title": "Global information",
          "name": "Name",
          "creation_date": "Creation date",
          "type": "Type",
          "nb_files": "Number of files",
          "files_total_size": "Total files size",
          "files_mean_size": "Average file size"
        },
        "files_list_tab": {
          "title": "Files list",
          "name": "Name",
          "type": "Type",
          "size": "Size",
          "action": "Action",
          "delete_action": "Delete"
        }
      }
    }
  }
}