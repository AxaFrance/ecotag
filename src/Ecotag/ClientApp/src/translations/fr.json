{
  "header": {
    "title": "Un écosystème qui vise à être écologique",
    "errors": {
      "not_connected": "Non connecté"
    }
  },
  "home": {
    "title": "Accueil",
    "projects": "Projets",
    "datasets": "Datasets",
    "groups": "Groupes",
    "errors": {
      "no_roles": "Aucun rôle n'est attribué à votre profil."
    }
  }
}