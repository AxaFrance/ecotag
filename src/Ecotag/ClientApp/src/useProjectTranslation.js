import './i18n';
import {useTranslation} from "react-i18next";
import i18next from 'i18next';

const useProjectTranslation = (namespace = 'translation') => {
    const {t: translate} = useTranslation(namespace);
    return {translate};
};

export const changeProjectTranslationLanguage = (language) => {
    i18next.changeLanguage(language);
};

export default useProjectTranslation;
