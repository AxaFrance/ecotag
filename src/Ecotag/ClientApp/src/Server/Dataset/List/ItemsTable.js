import React from "react";
import Table, {Paging} from '@axa-fr/react-toolkit-table';
import Loader from '@axa-fr/react-toolkit-loader';
import Action from "@axa-fr/react-toolkit-action";
import {useHistory} from "react-router-dom";
import {formatTimestampToString} from "../../date";
import {Locked} from "../Dataset.service";
import useProjectTranslation from "../../../translations/useProjectTranslation";

const getIconClassname = (locked) => {
    switch (locked) {
        case Locked.Locked:
            return 'btn af-btn--circle af-btn--locked';
        case Locked.LockedAndWorkInProgress:
            return 'btn af-btn--circle af-btn--locked-and-work-in-progress';
        case Locked.Pending:
            return 'btn af-btn--circle af-btn--pending';
        case Locked.None:
            return 'btn af-btn--circle af-btn--none';

    }
}

const getIcon = (locked) => {
    switch (locked) {
        case Locked.Locked:
        case Locked.LockedAndWorkInProgress:
            return 'lock';
        case Locked.Pending:
            return 'refresh';
        case Locked.None:
            return 'unlock';

    }
}

const ItemsTable = ({items, filters, loaderMode, onChangePaging}) => {

    const history = useHistory();
    const {translate} = useProjectTranslation();

    const editDatasetButton = id => {
        const path = `datasets/${id}`
        history.push(path);
    };

    const getTitle = (locked) => {
        switch (locked) {
            case Locked.Locked:
                return translate('dataset.list.actions_titles.locked');
            case Locked.LockedAndWorkInProgress:
                return translate('dataset.list.actions_titles.locked_and_work_in_progress');
            case Locked.Pending:
                return translate('dataset.list.actions_titles.pending');
            case Locked.None:
                return translate('dataset.list.actions_titles.none');
        }
    };

    return (
        <Loader mode={loaderMode}>
            <Table>
                <Table.Header>
                    <Table.Tr>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.lock')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.name')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.team')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.classification')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.nb_files')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.type')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('dataset.list.header.creation_date')}</span>
                        </Table.Th>
                        <Table.Th><span className="af-table__th-content">{translate('dataset.list.header.action')}</span></Table.Th>
                    </Table.Tr>
                </Table.Header>
                <Table.Body>
                    {items.map(
                        ({id, name, type, groupName, classification, numberFiles, createDate, locked}) => (
                            <Table.Tr key={id}>
                                <Table.Td>
                                    <Action className={getIconClassname(locked)}
                                            id="lock" icon={getIcon(locked)}
                                            title={getTitle(locked)}
                                            onClick={() => {
                                            }}/>
                                </Table.Td>
                                <Table.Td>{name}</Table.Td>
                                <Table.Td>{groupName}</Table.Td>
                                <Table.Td>{classification}</Table.Td>
                                <Table.Td>{numberFiles}</Table.Td>
                                <Table.Td>{type}</Table.Td>
                                <Table.Td>{formatTimestampToString(createDate)}</Table.Td>
                                <Table.Td>
                                    <Action id="id" icon="edit" title={translate('dataset.list.edit')} onClick={() => {
                                        editDatasetButton(id)
                                    }}/>
                                </Table.Td>
                            </Table.Tr>
                        )
                    )}
                </Table.Body>
            </Table>
            <Paging
                onChange={onChangePaging}
                numberItems={filters.paging.numberItemsByPage}
                numberPages={filters.paging.numberPages}
                currentPage={filters.paging.currentPage}
                id="home_paging"
                displayLabel={translate('paging.display')}
                elementsLabel={translate('paging.elements')}
                previousLabel={translate('paging.previous')}
                nextLabel={translate('paging.next')}
            />
        </Loader>

    )
};

export default ItemsTable;