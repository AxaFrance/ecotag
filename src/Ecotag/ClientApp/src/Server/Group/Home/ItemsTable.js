import React, {useState} from "react";
import Table, {Paging} from '@axa-fr/react-toolkit-table';
import Action from '@axa-fr/react-toolkit-action';
import Edit from "./Edit";
import './ItemsTable.scss';
import useProjectTranslation from "../../../useProjectTranslation";

const computeUsersList = users => {
    if (users.length === 0) {
        return "-";
    }
    return users.map(user => user.email).join(', ');
};

const UserRow = ({id, name, users, eligibleUsers, onUpdateUser}) => {
    const [isManageUsersModalVisible, setManageUsersModalVisible] = useState(false);
    const {translate} = useProjectTranslation();
    return (
        <Table.Tr key={id}>
            <Table.Td>{name}</Table.Td>
            <Table.Td>{computeUsersList(users)}</Table.Td>
            <Table.Td class="af-table__cell af-table__small">
                <Action id="editActionId" icon="pencil" title={translate('group.list.edit')}
                        onClick={() => setManageUsersModalVisible(true)}/>
                <Edit
                    idGroup={id}
                    users={users}
                    eligibleUsers={eligibleUsers}
                    onUpdateUser={onUpdateUser}
                    isManageUsersModalVisible={isManageUsersModalVisible}
                    setManageUsersModalVisible={setManageUsersModalVisible}
                />
            </Table.Td>
        </Table.Tr>
    );
};

const ItemsTable = ({items, filters, onChangePaging, onUpdateUser}) => {
    const {translate} = useProjectTranslation();

    return (
        <>
            <Table>
                <Table.Header>
                    <Table.Tr>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.name')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.users')}</span>
                        </Table.Th>
                        <Table.Th>
                            <span className="af-table__th-content">{translate('group.list.header.actions')}</span>
                        </Table.Th>
                    </Table.Tr>
                </Table.Header>
                <Table.Body>
                    {items.map(({id, name, users = [], eligibleUsers = []}) => (
                        <UserRow
                            key={id}
                            id={id}
                            name={name}
                            users={users}
                            eligibleUsers={eligibleUsers}
                            onUpdateUser={onUpdateUser}
                        />
                    ))}
                </Table.Body>
            </Table>
            <Paging
                onChange={onChangePaging}
                numberItems={filters.paging.numberItemsByPage}
                numberPages={filters.paging.numberPages}
                currentPage={filters.paging.currentPage}
                id="home_paging"
                displayLabel={translate('group.paging.display')}
                elementsLabel={translate('group.paging.elements')}
                previousLabel={translate('group.paging.previous')}
                nextLabel={translate('group.paging.next')}
            />
        </>
    )
};

export default ItemsTable;