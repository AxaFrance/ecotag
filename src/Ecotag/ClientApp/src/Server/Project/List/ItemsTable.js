import HeaderColumnCell from './ColumnHeader';
import React, {useEffect, useState} from 'react';
import {useHistory} from 'react-router-dom';
import Table, {Paging} from '@axa-fr/react-toolkit-table';
import Action from '@axa-fr/react-toolkit-action';
import {formatTimestampToString} from '../../date';
import {fetchAnnotationsStatus} from '../Project.service';
import {resilienceStatus} from '../../shared/Resilience';
import {Popover} from '@axa-fr/react-toolkit-all';
import {ProgressBar} from './ProgressBar';
import useProjectTranslation from '../../../translations/useProjectTranslation';

const NumberTagToDo = ({state}) => {
    const {translate} = useProjectTranslation();
    const {ERROR, SUCCESS, LOADING} = resilienceStatus;
    const annotationsStatus = state.annotationsStatus;
    return <>{{
        [LOADING]: <span>{translate('project.projects_list.list.nb_tags_to_do.resilience.loading')}</span>,
        [ERROR]: (
            <span>{translate('project.projects_list.list.nb_tags_to_do.resilience.error')}</span>
        ),
        [SUCCESS]: <>
            <div className="projects-af-popover__wrapper">
                <Popover
                    placement="top"
                    classModifier="project-home"
                >
                    <Popover.Pop>
                        <h3>{translate('project.projects_list.list.nb_tags_to_do.title')}</h3>
                        <table>
                            <tr>
                                <td>{translate('project.projects_list.list.nb_tags_to_do.to_do')}</td>
                                <td>{translate('project.projects_list.list.nb_tags_to_do.done')}</td>
                                <td>{translate('project.projects_list.list.nb_tags_to_do.remaining')}</td>
                            </tr>
                            <tr>
                                <td>{annotationsStatus.numberAnnotationsToDo}</td>
                                <td>{annotationsStatus.numberAnnotationsDone}</td>
                                <td>{annotationsStatus.numberAnnotationsToDo - annotationsStatus.numberAnnotationsDone}</td>
                            </tr>
                        </table>
                    </Popover.Pop>
                    <Popover.Over>
                        <ProgressBar percentage={annotationsStatus.percentageNumberAnnotationsDone}
                                     label={annotationsStatus.percentageNumberAnnotationsDone === 100 ?
                                         translate('project.projects_list.list.nb_tags_to_do.percentage.done') :
                                         translate('project.projects_list.list.nb_tags_to_do.percentage.pending')}
                        />
                    </Popover.Over>
                </Popover>
            </div>
        </>
    }[state.status]
    }
    </>

}


const initAsync = (fetch) => async (state, setState, id) => {

    const annotationsStatusResponse = await fetchAnnotationsStatus(fetch)(id);
    if (annotationsStatusResponse.status >= 500 || annotationsStatusResponse.status === 403) {
        const data = {
            ...state,
            status: annotationsStatusResponse.status === 403 ? resilienceStatus.FORBIDDEN : resilienceStatus.ERROR
        };
        setState(data);
    } else {
        const annotationsStatus = await annotationsStatusResponse.json();
        const data = {annotationsStatus, status: resilienceStatus.SUCCESS};
        setState(data);
    }

}

const ProjectRow = ({id, name, groupName, createDate, annotationType, fetch}) => {
    const history = useHistory();
    const {translate} = useProjectTranslation();
    const projectPageButton = id => {
        const path = `/projects/${id}`;
        history.push(path);
    };
    const [state, setState] = useState({
        annotationsStatus: {
            isAnnotationClosed: true,
            numberAnnotationsByUsers: [],
            numberAnnotationsDone: 0,
            numberAnnotationsToDo: 0,
            percentageNumberAnnotationsDone: 0
        },
        status: resilienceStatus.LOADING
    })
    useEffect(async () => {
        initAsync(fetch)(state, setState, id);
    }, []);

    return (
        <Table.Tr key={id}>
            <Table.Td>{name}</Table.Td>
            <Table.Td>{groupName}</Table.Td>
            <Table.Td>{createDate}</Table.Td>
            <Table.Td>{annotationType}</Table.Td>
            <Table.Td>
                <NumberTagToDo state={state}/>
            </Table.Td>
            <Table.Td>
                <Action id="id" icon="zoom-in" title={translate('project.projects_list.list.headers.edit')} onClick={() => projectPageButton(id)}/>
            </Table.Td>
        </Table.Tr>
    );
};

const ItemsTable = ({items, filters, onChangePaging, onChangeSort, fetch}) => {
    const {translate} = useProjectTranslation();

    return (
        <>
            <Table>
                <Table.Header>
                    <Table.Tr>
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('name')}
                            headerColumnName={translate('project.projects_list.list.headers.name')}
                            filterColumnValue={filters.columns.name.value}
                        />
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('groupName')}
                            headerColumnName={translate('project.projects_list.list.headers.team')}
                            filterColumnValue={filters.columns.groupName.value}
                        />
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('createDate')}
                            headerColumnName={translate('project.projects_list.list.headers.creation_date')}
                            filterColumnValue={filters.columns.createDate.value}
                        />
                        <HeaderColumnCell
                            onChangeSort={onChangeSort('annotationType')}
                            headerColumnName={translate('project.projects_list.list.headers.annotation_type')}
                            filterColumnValue={filters.columns.annotationType.value}
                        />
                        <HeaderColumnCell
                            headerColumnName={translate('project.projects_list.list.headers.status')}
                        />
                        <Table.Th classModifier="sortable">
                            <span className="af-table__th-content af-btn__text">{translate('project.projects_list.list.headers.action')}</span>
                        </Table.Th>
                    </Table.Tr>
                </Table.Header>
                <Table.Body>
                    {items.map(({id, name, groupName, createDate, numberTagToDo, annotationType}) => (
                        <ProjectRow
                            key={id}
                            id={id}
                            name={name}
                            groupName={groupName}
                            createDate={formatTimestampToString(createDate)}
                            numberTagToDo={numberTagToDo}
                            annotationType={annotationType}
                            fetch={fetch}
                        />
                    ))}
                </Table.Body>
            </Table>
            <Paging
                onChange={onChangePaging}
                numberItems={filters.paging.numberItemsByPage}
                numberPages={filters.paging.numberPages}
                currentPage={filters.paging.currentPage}
                id="home_paging"
                displayLabel={translate('paging.display')}
                elementsLabel={translate('paging.elements')}
                previousLabel={translate('paging.previous')}
                nextLabel={translate('paging.next')}
            />
        </>
    )

};

export default ItemsTable;