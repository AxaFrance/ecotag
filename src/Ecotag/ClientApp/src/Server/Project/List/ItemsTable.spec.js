import {render} from "@testing-library/react";
import {LoaderModes} from "@axa-fr/react-toolkit-all";
import ItemsTable from "./ItemsTable";
import {changeProjectTranslationLanguage} from "../../../translations/useProjectTranslation";

const items = [{
    id: "0001",
    name: "Green card",
    groupName: "groupName",
    annotationType: "Image",
    numberTagToDo: 300,
    createDate: new Date("10-30-2019").getTime()
}];

const filters = {
    paging: {
        numberItemsByPage: 10,
        currentPage: 1,
    },
    filterValue: null,
    columns: {
        name: {value: null, timeLastUpdate: null},
        groupName: {value: null, timeLastUpdate: null},
        createDate: {value: 'desc', timeLastUpdate: new Date()},
        annotationType: {value: null, timeLastUpdate: null},
        numberTagToDo: {value: null, timeLastUpdate: null}
    },
};

export const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));

export const fetch = async (url, config) => {
    await sleep(1);
    return {
        status: 201,
        json: async () => {
            await sleep(1);
            return {};
        }
    };
};

describe("Check Project ItemsTable behaviour", () => {
    const renderComponentAndCheckSnapshot = async() => {
        const {asFragment} = render(<ItemsTable items={items} filters={filters} loaderMode={LoaderModes.none}
                                                onChangePaging={() => {
                                                }} onChangeSort={() => {
        }} fetch={fetch}/>);
        expect(asFragment()).toMatchSnapshot();
    };
    it("should render correctly in english", async () => {
        changeProjectTranslationLanguage('en');
        await renderComponentAndCheckSnapshot();
    });
    it('should render correctly in french', async() => {
        changeProjectTranslationLanguage('fr');
        await renderComponentAndCheckSnapshot();
    });
});
