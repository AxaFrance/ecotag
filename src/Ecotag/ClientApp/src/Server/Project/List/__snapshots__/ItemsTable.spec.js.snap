// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Project ItemsTable behaviour should render correctly in english 1`] = `
<DocumentFragment>
  <table
    class="af-table"
  >
    <thead
      class="af-table-thead"
    >
      <tr
        class="af-table__tr"
      >
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Name
              </span>
              <i
                class="glyphicon glyphicon-sorting"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Team
              </span>
              <i
                class="glyphicon glyphicon-sorting"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Creation date
              </span>
              <i
                class="glyphicon glyphicon-sorting-desc"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Annotation type
              </span>
              <i
                class="glyphicon glyphicon-sorting"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Status
              </span>
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content af-btn__text"
          >
            Action
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      class="af-table__body"
    >
      <tr
        class="af-table__tr"
      >
        <td
          class="af-table__cell"
        >
          Green card
        </td>
        <td
          class="af-table__cell"
        >
          groupName
        </td>
        <td
          class="af-table__cell"
        >
          02/01/0001
        </td>
        <td
          class="af-table__cell"
        >
          Image
        </td>
        <td
          class="af-table__cell"
        >
          <span>
            Loading...
          </span>
        </td>
        <td
          class="af-table__cell"
        >
          <a
            class="btn af-btn--circle"
            href="#"
            id="id"
            role="button"
            tabindex="0"
            title="Edit"
          >
            <i
              class="glyphicon glyphicon-zoom-in"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="af-paging"
  >
    <div
      class="af-paging__limit"
    >
      <div
        class="af-paging__limit"
      >
        <form
          class="af-paging__form"
        >
          <div
            class="af-form__group"
          >
            <div
              class="col col-sm-2 col-md-2 col-lg-2 col-xl-2"
            >
              <label
                class="af-form__group-label"
                for="home_paging"
              >
                Display
              </label>
            </div>
            <div
              class="col col-sm-10 col-md-10 col-lg-10 col-xl-10"
            >
              <div
                class="af-form__select"
              >
                <div
                  class="af-form__select-container"
                >
                  <select
                    class="af-form__input-select"
                    id="home_paging"
                  >
                    <option
                      value="5"
                    >
                      5
                    </option>
                    <option
                      value="10"
                    >
                      10
                    </option>
                    <option
                      value="25"
                    >
                      25
                    </option>
                    <option
                      value="50"
                    >
                      50
                    </option>
                    <option
                      value="100"
                    >
                      100
                    </option>
                  </select>
                  <span
                    class="glyphicon glyphicon-menu-down"
                  />
                </div>
                <span
                  class="af-form__input-cmplt"
                >
                  elements
                </span>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="af-paging__pager"
    >
      <nav
        class="af-pager"
      >
        <ul
          class="af-pager__pagination"
        >
          <li
            class="af-pager__item af-pager__item--disabled"
          >
            <span
              class="af-pager__item-nolink"
            >
              « Previous
            </span>
          </li>
          <li
            class="af-pager__item af-pager__item--active"
          >
            <a
              class="af-pager__item-link"
              href="#"
            >
              <span>
                1
              </span>
            </a>
          </li>
          <li
            class="af-pager__item af-pager__item--disabled"
          >
            <span
              class="af-pager__item-nolink"
            >
              Next »
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Check Project ItemsTable behaviour should render correctly in french 1`] = `
<DocumentFragment>
  <table
    class="af-table"
  >
    <thead
      class="af-table-thead"
    >
      <tr
        class="af-table__tr"
      >
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Nom
              </span>
              <i
                class="glyphicon glyphicon-sorting"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Équipe
              </span>
              <i
                class="glyphicon glyphicon-sorting"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Date de création
              </span>
              <i
                class="glyphicon glyphicon-sorting-desc"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Type d'annotation
              </span>
              <i
                class="glyphicon glyphicon-sorting"
              />
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content"
          >
            <button
              class="af-btn af-btn--table-sorting"
            >
              <span
                class="af-btn__text"
              >
                Statut
              </span>
            </button>
          </span>
        </th>
        <th
          class="af-table__th af-table__th--sortable"
          classmodifier="sortable"
        >
          <span
            class="af-table__th-content af-btn__text"
          >
            Action
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      class="af-table__body"
    >
      <tr
        class="af-table__tr"
      >
        <td
          class="af-table__cell"
        >
          Green card
        </td>
        <td
          class="af-table__cell"
        >
          groupName
        </td>
        <td
          class="af-table__cell"
        >
          02/01/0001
        </td>
        <td
          class="af-table__cell"
        >
          Image
        </td>
        <td
          class="af-table__cell"
        >
          <span>
            Chargement...
          </span>
        </td>
        <td
          class="af-table__cell"
        >
          <a
            class="btn af-btn--circle"
            href="#"
            id="id"
            role="button"
            tabindex="0"
            title="Éditer"
          >
            <i
              class="glyphicon glyphicon-zoom-in"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="af-paging"
  >
    <div
      class="af-paging__limit"
    >
      <div
        class="af-paging__limit"
      >
        <form
          class="af-paging__form"
        >
          <div
            class="af-form__group"
          >
            <div
              class="col col-sm-2 col-md-2 col-lg-2 col-xl-2"
            >
              <label
                class="af-form__group-label"
                for="home_paging"
              >
                Afficher
              </label>
            </div>
            <div
              class="col col-sm-10 col-md-10 col-lg-10 col-xl-10"
            >
              <div
                class="af-form__select"
              >
                <div
                  class="af-form__select-container"
                >
                  <select
                    class="af-form__input-select"
                    id="home_paging"
                  >
                    <option
                      value="5"
                    >
                      5
                    </option>
                    <option
                      value="10"
                    >
                      10
                    </option>
                    <option
                      value="25"
                    >
                      25
                    </option>
                    <option
                      value="50"
                    >
                      50
                    </option>
                    <option
                      value="100"
                    >
                      100
                    </option>
                  </select>
                  <span
                    class="glyphicon glyphicon-menu-down"
                  />
                </div>
                <span
                  class="af-form__input-cmplt"
                >
                  éléments
                </span>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="af-paging__pager"
    >
      <nav
        class="af-pager"
      >
        <ul
          class="af-pager__pagination"
        >
          <li
            class="af-pager__item af-pager__item--disabled"
          >
            <span
              class="af-pager__item-nolink"
            >
              « Précédent
            </span>
          </li>
          <li
            class="af-pager__item af-pager__item--active"
          >
            <a
              class="af-pager__item-link"
              href="#"
            >
              <span>
                1
              </span>
            </a>
          </li>
          <li
            class="af-pager__item af-pager__item--disabled"
          >
            <span
              class="af-pager__item-nolink"
            >
              Suivant »
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
